File: python/generador.py
'''generador de tablas de multiplicar'''

import utilidades as ut

def main():
    ut.linea(50)
    print("Bienvenido al generador de tablas de multiplicar".upper())
    num : int = 0
    inicio : int = 0
    final : int = 0
    
    ut.encabezado('generador de tablas', 50)
    num = ut.leerNumero('ingresa tu numero', int)
    ut.encabezado(f'tabla de multiplicar del {num}', 50)
    generarTabla(num)
    ut.encabezado('generar rango de tablas de multiplicar', 50)
    inicio = ut.leerNumero('ingresa el numero de inicio', int)
    final = ut.leerNumero('ingresa el numero de final', int)
    ut.encabezado(f'tabla de multiplicar del {inicio} al {final}', 50)
    generarRangoTablas(inicio, final)
    
def generarRangoTablas(inicio: int, final: int):
    nroTabla: int = abs( inicio - final ) + 1
    for i in range(nroTabla):
        for j in range (1, 11):
            print(f'{inicio} X {j} = {producto(inicio, j)}')
        ut.linea(50)
        inicio += 1
        
    
    
    
def generarTabla(num: int) -> None:
    for i in range(1,11):
        print(f'{num} x {i} = {producto(num, i)}')
        
def producto(a: int, b: int) -> int:
    return a * b
        
    
main()

File: JS/src/generador.js
/* ganerador de tablas de multi */
import { encabezado, leerNumero, linea} from "./utilidades";

const main = () => {
    //definir variables
    let numero = 0
    let inicio = 0
    let final  = 0
    
    encabezado("generador de tablas de multiplicar", 50)
    numero = leerNumero("ingresar tu numero ")
    encabezado(`tabla del ${numero}`, 50)
    generarTablas(numero)
    encabezado("generador de rango de tablas", 50)
    inicio = leerNumero("Ingrese numero de inicio")
    final = leerNumero("ingrese numero final")
    encabezado(`tablas de multiplicar del ${inicio} al ${final}`, 50)
    generarRangoTablas(inicio, final)
}

const generarRangoTablas = function(inicio = 0, final = 0) {
    let nroTablas = Math.abs(inicio - final) +1
    for (let i = 0; i < nroTablas; i++) {
       for (let j = 0; j < 11; j++) {
      console.log(`${inicio} X ${j} = ${producto(inicio, j)}`)
       }
       linea(50)
       inicio++
    }
}


const generarTablas = function (numero = 0) {
    for (let i = 1; i < 11; i++){
        console.log(`${numero} X ${i} = ${producto(numero, i)}`)
    }
}

const producto = (a=0, b=0) => a * b;


main()